R/ResampleResult.R

```diff
@@ -80,4 +82,128 @@ fortify.ResampleResult = function(model, data, measure = NULL, ...) {
+
+  if (length(features) != 2) {
+    stop("plot_learner_prediction.TaskClassif only works for tasks with two features!")
+  }
+
+  grid_res = predict_grid(learners, task, grid_points = grid_points, expand_range = expand_range)
+
+  if (length(learners) > 1) {
+    facet_labels = setNames(paste(object$resampling$id, seq_len(object$resampling$iters)), seq_len(object$resampling$iters))
+    folds_facet = facet_wrap(".id", labeller = as_labeller(facet_labels))
+  } else {
+    folds_facet = NULL
+  }
+
+  if (learner$predict_type == "prob") {
+    grid_res[, "prob.response" := .SD[, paste0("prob.", get("response")), with = FALSE] , by = "response"]
+    raster_aes = aes_string(fill = "response", alpha = "prob.response")
+  } else {
+    raster_aes = aes_string(fill = "response")
+  }
+
+  ggplot(grid_res, aes_string(features[1L], features[2L])) +
+    geom_raster(raster_aes) +
+    geom_point(data = task_data(object), aes_string(features[1L], features[2L], fill = task$target_names, shape = ".point_type"), size = 4, color = "black") +
+    scale_shape_manual(values = c(train = 21, test = 22, both = 23), name = "set") +
+    folds_facet
+}
+
+plot_learner_prediction_resample_result.LearnerRegr = function(learner, object, measure, grid_points = 100L, expand_range = 0) {
+
+  task = object$task
+  features = task$feature_names
+  learners = object$learners
+
+  if (length(features) > 2) {
+    stop("plot_learner_prediction.LearnerRegr only works with one or two features!")
+  }
+
+  grid = predict_grid(learners, task, grid_points, expand_range)
+
+  if (length(learners) > 1) {
+    facet_labels = setNames(paste(object$resampling$id, seq_len(object$resampling$iters)), seq_len(object$resampling$iters))
+    folds_facet = facet_wrap(".id", labeller = as_labeller(facet_labels))
+  } else {
+    folds_facet = NULL
+  }
+
+  if (length(features) == 1) {
+    if (learner$predict_type == "se") {
+      se_geom = geom_ribbon(aes_string(ymin = "response-se", ymax = "response+se"), alpha = 0.2)
+    } else {
+      se_geom = NULL
+    }
+    g = ggplot(grid, aes_string(features, "response")) +
+      se_geom +
+      geom_line() +
+      geom_point(data = task_data(object), aes_string(features, task$target_names, shape = ".point_type", color = ".point_type"), size = 4) +
+      scale_shape_manual(values = c(train = 16, test = 15, both = 17), name = "set") +
+      folds_facet
+  } else if (length(features) == 2) {
+    g = ggplot(grid, aes_string(features[1L], features[2L])) +
+      geom_raster(aes_string(fill = "response")) +
+      geom_point(data = task_data(object), aes_string(features[1L], features[2L], fill = task$target_names, shape = ".point_type"), size = 4, color = "black", shape = 21) +
+      scale_shape_manual(values = c(train = 21, test = 22, both = 23), name = "set", shape = 21) +
+      folds_facet
+  }
+
+  return(g)
+
+}
+
+task_data = function(object) {
+  data = lapply(seq_along(object$learners), function(i) {
+    data = object$task$data()
```

Comment:
What about observation which are neither used in train nor in test?

Reply:
good point. we do not plot them?